artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java

```diff
@@ -290,9 +292,19 @@ public int getServerVersion() {
 
    @Override
    public ClientSession.AddressQuery addressQuery(final SimpleString address) throws ActiveMQException {
-      SessionBindingQueryResponseMessage_V3 response = (SessionBindingQueryResponseMessage_V3) sessionChannel.sendBlocking(new SessionBindingQueryMessage(address), PacketImpl.SESS_BINDINGQUERY_RESP_V3);
-
-      return new AddressQueryImpl(response.isExists(), response.getQueueNames(), response.isAutoCreateJmsQueues(), response.isAutoCreateJmsTopics());
+      if (sessionChannel.supports(PacketImpl.SESS_BINDINGQUERY_RESP_V3)) {
```

Comment:
```sessionChannel.supports()``` doesn't give correct results, I think I was trying that. The problem was that ```connection.getClientVersion()``` returned zero in ```ChannelImpl.supports()```, hence I was comparing version numbers directly.